src/Parsers/ASTSelectQuery.h

```diff
@@ -89,6 +89,8 @@ class ASTSelectQuery : public IAST
     bool group_by_with_grouping_sets = false;
     bool limit_with_ties = false;
 
+    bool query_normalized = false;
```

Comment:
Yes, I also found it is not correct, but I coundn't find a proper way to fix it.

Reply:
FWIW the behavior has been changed a little in https://github.com/ClickHouse/ClickHouse/pull/28746/files#diff-03665017349505c5529e5d0eeeeae80081e064267d6035b94cd428baf11856ceR1044-R1045

Reply:
This code seems does not exists anymore.

Reply:
We should rewrite columns in which places?

Reply:
We should rewrite columns in which places? I don't understand how it works.

Reply:
`select toString(dummy) dummy from remote('127.{1,2}',system,one) group by dummy` -> `select toString(dummy) dummy from remote('127.{1,2}',system,one) a group by toString(dummy)` -> `select toString(dummy) dummy from remote('127.{1,2}',system,one) a group by a.dummy`

So after removing injective function, we need to use qualified name, to avoid using alias.